service/matching/taskWriter.go

```diff
@@ -73,23 +73,19 @@ var errShutdown = &persistence.ConditionFailedError{Msg: "task queue shutting do
 
 func newTaskWriter(
 	tlMgr *taskQueueManagerImpl,
-	block taskIDBlock,
 ) *taskWriter {
 	return &taskWriter{
 		status:       common.DaemonStatusInitialized,
 		tlMgr:        tlMgr,
 		config:       tlMgr.config,
 		taskQueueID:  tlMgr.taskQueueID,
 		appendCh:     make(chan *writeTaskRequest, tlMgr.config.OutstandingTaskAppendsThreshold()),
-		taskIDBlock:  block,
-		maxReadLevel: block.start - 1,
 		logger:       tlMgr.logger,
-
 		shutdownChan: make(chan struct{}),
 	}
 }
 
-func (w *taskWriter) Start() {
+func (w *taskWriter) Start(initialBlock taskIDBlock) {
```

Comment:
does initialBlock need to be input parameter? can it be an internal var?

Reply:
It can if we delay the instantiation of the taskWriter into the Start() function in the enclosing taskQueueManagerImpl. My preference is to separate construction from "doing stuff" which should only occur when Start() is called. I include performing database lookups to obtain an id block (which can preempt other remote owners) as "doing stuff"

Reply:
as temp fix, i do not care much, but long term start / stop function should in general not take in any parameters, all inputs should be initialized within struct initialization

Reply:
We should talk about requiring parameterless Start() functions. I'm not sure the restriction is worthwhile. In this case, that value requires a database lookup - not something you want to do at construction-time.

Reply:
> We should talk about requiring parameterless Start() functions. I'm not sure the restriction is worthwhile. In this case, that value requires a database lookup - not something you want to do at construction-time.

depends, either do DB lookup prior to construction & pass in for struct initialization or during construction, do DB lookup.